Source/Core/EllipsoidRhumbLine.js

```diff
@@ -0,0 +1,533 @@
+                return this._end;
+            }
+        },
+
+        /**
+         * Gets the heading from the start point to the end point.
+         * @memberof EllipsoidRhumbLine.prototype
+         * @type {Number}
+         * @readonly
+         */
+        heading : {
+            get : function() {
+                //>>includeStart('debug', pragmas.debug);
+                Check.defined('distance', this._distance);
+                //>>includeEnd('debug');
+
+                return this._heading;
+            }
+        }
+    });
+
+    /**
+     * Create a rhumb line using an initial and final position.
+     *
+     * @param {Cartographic} start The initial planetodetic point on the path.
+     * @param {Cartographic} end The final planetodetic point on the path.
+     * @param {Ellipsoid} [ellipsoid=Ellipsoid.WGS84] The ellipsoid on which the rhumb line lies.
+     * @param {EllipsoidRhumbLine} [result] The object in which to store the result.
+     * @returns {EllipsoidRhumbLine} The EllipsoidRhumbLine object.
+     */
+    EllipsoidRhumbLine.fromStartAndEnd = function(start, end, ellipsoid, result) {
+        //>>includeStart('debug', pragmas.debug);
+        Check.defined('start', start);
+        Check.defined('end', end);
+        //>>includeEnd('debug');
+
+        var e = defaultValue(ellipsoid, Ellipsoid.WGS84);
+
+        if (defined(result)) {
+            result._ellipsoid = e;
+            result.setEndPoints(start, end);
+            return result;
+        }
+        return new EllipsoidRhumbLine(start, end, e);
+    };
+
+    /**
+     * Create a rhumb line using an initial position with a heading and distance.
+     *
+     * @param {Cartographic} start The initial planetodetic point on the path.
+     * @param {Number} heading The heading in radians.
+     * @param {Number} distance The rhumb line distance between the start and end point.
+     * @param {Ellipsoid} [ellipsoid=Ellipsoid.WGS84] The ellipsoid on which the rhumb line lies.
+     * @param {EllipsoidRhumbLine} [result] The object in which to store the result.
+     * @returns {EllipsoidRhumbLine} The EllipsoidRhumbLine object.
+     */
+    EllipsoidRhumbLine.fromStartHeadingDistance = function(start, heading, distance, ellipsoid, result) {
+        //>>includeStart('debug', pragmas.debug);
+        Check.defined('start', start);
+        Check.defined('heading', heading);
+        Check.defined('distance', distance);
+        Check.typeOf.number.greaterThan('distance', distance, 0.0);
+        //>>includeEnd('debug');
+
+        var e = defaultValue(ellipsoid, Ellipsoid.WGS84);
+
+        if (!defined(result)) {
+            result = new EllipsoidRhumbLine(undefined, undefined, e);
+        }
+
+        initialize(result, start, undefined, e);
+        result._heading = CesiumMath.negativePiToPi(heading);
+        result._distance = distance;
+
+        result._end = result.interpolateUsingSurfaceDistance(distance, new Cartographic());
```

Comment:
This whole function is kind of an odd implementation to me.  Why not break out `interpolateUsingSurfaceDistance` into a standalone local private function that can be called with `start, heading, distance, ellipsoid`, us that to compute the end point then just create the fully contructed result.  Using a half-initialized instance like this seems like something that could bite us.

Obviously we still expose the prototype version of `interpolateUsingSurfaceDistance` that just called the local one.

Reply:
Yeah, this could work better. I'll make the change.